builtin/logical/mysql/path_config_connection.go

```diff
@@ -32,8 +36,14 @@ func (b *backend) pathConnectionWrite(
 	req *logical.Request, data *framework.FieldData) (*logical.Response, error) {
 	connString := data.Get("value").(string)
 
+	maxOpenConns := data.Get("max_open_connections").(int)
+	if maxOpenConns == 0 {
+		maxOpenConns = 2
+	}
```

Comment:
@jefferai  If I understand this correctly, if max_open_connection is passed as a parameter and neither the connectionString nor the connectionURL is specified, we are retaining what was previously written and modifying just the max_open_connections?

It is always assumed that a new entry is a fresh entry and this entails that the ignored parameters take the default values (in all the APIs of Vault). 

If we change it here, will it not be a confusing UX? OR Are we planning to change it to reflect this UX all the way, going forward?